Traceback (most recent call last):
  File "C:\Users\pm27995\AppData\Local\anaconda3\envs\qe-mini-example\lib\site-packages\jupyter_cache\executors\utils.py", line 58, in single_nb_execution
    executenb(
  File "C:\Users\pm27995\AppData\Local\anaconda3\envs\qe-mini-example\lib\site-packages\nbclient\client.py", line 1305, in execute
    return NotebookClient(nb=nb, resources=resources, km=km, **kwargs).execute()
  File "C:\Users\pm27995\AppData\Local\anaconda3\envs\qe-mini-example\lib\site-packages\jupyter_core\utils\__init__.py", line 165, in wrapped
    return loop.run_until_complete(inner)
  File "C:\Users\pm27995\AppData\Local\anaconda3\envs\qe-mini-example\lib\asyncio\base_events.py", line 616, in run_until_complete
    return future.result()
  File "C:\Users\pm27995\AppData\Local\anaconda3\envs\qe-mini-example\lib\site-packages\nbclient\client.py", line 705, in async_execute
    await self.async_execute_cell(
  File "C:\Users\pm27995\AppData\Local\anaconda3\envs\qe-mini-example\lib\site-packages\nbclient\client.py", line 1058, in async_execute_cell
    await self._check_raise_for_error(cell, cell_index, exec_reply)
  File "C:\Users\pm27995\AppData\Local\anaconda3\envs\qe-mini-example\lib\site-packages\nbclient\client.py", line 914, in _check_raise_for_error
    raise CellExecutionError.from_cell_and_msg(cell, exec_reply_content)
nbclient.exceptions.CellExecutionError: An error occurred while executing the following cell:
------------------
L = 10000
mean = geostats.bootstrap(df['Porosity'].values,stat = np.average,weights = df['Wts'].values,nreal = L)
stdev = geostats.bootstrap(df['Porosity'].values,stat = np.std,weights = df['Wts'].values,nreal = L)

plt.subplot(121)
GSLIB.hist_st(mean,0.11,0.15,False,False,50,None,'Average Porosity (fraction)','Bootstrap Uncertainty in Porosity Average')

plt.subplot(122)
GSLIB.hist_st(stdev,0.015,0.045,False,False,50,None,'Standard Deviation Porosity (fraction)','Bootstrap Uncertainty in Porosity Standard Deviation')

plt.subplots_adjust(left=0.0, bottom=0.0, right=3.0, top=1.2, wspace=0.2, hspace=0.2)
plt.show()   
    
print('Summary Statistics for Bootstrap Porosity Mean Realizations:')
print(stats.describe(mean))
print('P10: ' + str(round(np.percentile(mean,10),3)) + ', P50: ' + str(round(np.percentile(mean,50),3)) + ', P90: ' + str(round(np.percentile(mean,90),3))) 

print('\nSummary Statistics for Bootstrap Porosity Standard Deviation Realizations:')
print(stats.describe(stdev))
print('P10: ' + str(round(np.percentile(stdev,10),3)) + ', P50: ' + str(round(np.percentile(stdev,50),3)) + ', P90: ' + str(round(np.percentile(stdev,90),3))) 
------------------


[1;31m---------------------------------------------------------------------------[0m
[1;31mValueError[0m                                Traceback (most recent call last)
Cell [1;32mIn[17], line 2[0m
[0;32m      1[0m L [38;5;241m=[39m [38;5;241m10000[39m
[1;32m----> 2[0m mean [38;5;241m=[39m [43mgeostats[49m[38;5;241;43m.[39;49m[43mbootstrap[49m[43m([49m[43mdf[49m[43m[[49m[38;5;124;43m'[39;49m[38;5;124;43mPorosity[39;49m[38;5;124;43m'[39;49m[43m][49m[38;5;241;43m.[39;49m[43mvalues[49m[43m,[49m[43mstat[49m[43m [49m[38;5;241;43m=[39;49m[43m [49m[43mnp[49m[38;5;241;43m.[39;49m[43maverage[49m[43m,[49m[43mweights[49m[43m [49m[38;5;241;43m=[39;49m[43m [49m[43mdf[49m[43m[[49m[38;5;124;43m'[39;49m[38;5;124;43mWts[39;49m[38;5;124;43m'[39;49m[43m][49m[38;5;241;43m.[39;49m[43mvalues[49m[43m,[49m[43mnreal[49m[43m [49m[38;5;241;43m=[39;49m[43m [49m[43mL[49m[43m)[49m
[0;32m      3[0m stdev [38;5;241m=[39m geostats[38;5;241m.[39mbootstrap(df[[38;5;124m'[39m[38;5;124mPorosity[39m[38;5;124m'[39m][38;5;241m.[39mvalues,stat [38;5;241m=[39m np[38;5;241m.[39mstd,weights [38;5;241m=[39m df[[38;5;124m'[39m[38;5;124mWts[39m[38;5;124m'[39m][38;5;241m.[39mvalues,nreal [38;5;241m=[39m L)
[0;32m      5[0m plt[38;5;241m.[39msubplot([38;5;241m121[39m)

File [1;32m~\AppData\Local\anaconda3\envs\qe-mini-example\lib\site-packages\geostatspy\geostats.py:5962[0m, in [0;36mbootstrap[1;34m(zdata, weights, nreal, stat)[0m
[0;32m   5960[0m [38;5;28;01mdef[39;00m [38;5;21mbootstrap[39m(zdata,weights,nreal,stat):
[0;32m   5961[0m     zreal [38;5;241m=[39m np[38;5;241m.[39mzeros(nreal)               [38;5;66;03m# declare an empty list to store the bootstrap realizations[39;00m
[1;32m-> 5962[0m     [38;5;28;01mif[39;00m [43mweights[49m[43m [49m[38;5;241;43m==[39;49m[43m [49m[38;5;28;43;01mNone[39;49;00m:
[0;32m   5963[0m         weights [38;5;241m=[39m np[38;5;241m.[39mones(([38;5;28mlen[39m(zdata)))
[0;32m   5964[0m     [38;5;28;01mfor[39;00m l [38;5;129;01min[39;00m [38;5;28mrange[39m([38;5;241m0[39m,nreal):              [38;5;66;03m# loop over the L bootstrap realizations[39;00m

[1;31mValueError[0m: The truth value of an array with more than one element is ambiguous. Use a.any() or a.all()

